gusto/timeloop.py

```diff
@@ -174,6 +191,15 @@ def run(self, t, tmax, x_end=None):
                 # advects field
                 advection.apply(field, field)
 
+            for physics in self.physics_list:
+                physics.apply()
+         
+	    #Added by PB, 15/05/2017
+	    with timed_stage("Diffusion"):
+              for name, diffusion in self.diffusion_dict.iteritems():
+                 field = getattr(state.fields, name)
+                 diffusion.apply(field, field)
```

Comment:
So now the advection time stepping scheme can include diffusion as well.  That seems... odd?

I wonder if we should think about how to redesign the way the time stepping is glued together.  Right now, if you want to do "advection + some stuff" you need to copy the full time loop, excise the bits you don't want, which is not a scalable method of writing code.

What if you want to compute diagnostics of various fields